Posts/2021/03/Caps-lock-ToUpper.md

```diff
@@ -0,0 +1,206 @@
+---
+post_title: Can I Enable the Caps Lock Key?
+username: tfl@psp.co.uk
+Catagories: PowerShell
+tags: Caps Lock, string, ToUpper()
+Summary: How to enable all upper case input
+---
+
+**Q:** I have a script where users enter some information.
+This information needs to be entered in all capital letters, so my instructions say, “Please make sure the Caps Lock key is on before entering the information.”
+They don’t always do that, however.
+Is there a way to turn the Caps Lock key on and off using a script?
+
+**A:**  I don't know how to run the key off and on, but with PowerShell, there is a way to mimic the effect of turning on the Caps Lock key.
+
+## User Input Considered Harmful
+
+Let's start with the observation that all user input is harmful.
+One of my earliest IT heroes, Edsger Dijkstra, published a seminal letter [Go To Statement Considered Harmful](https://homepages.cwi.nl/~storm/teaching/reader/Dijkstra68.pdf) in 1968 which began the structured programming revolution.
```

Comment:
```suggestion
One of my earliest IT heroes, Edgar Dijkstra, published a seminal letter [Go To Statement Considered Harmful](https://homepages.cwi.nl/~storm/teaching/reader/Dijkstra68.pdf) in 1968 which began the structured programming revolution.
```

Reply:
Got it.  The ACM paper had it as "Edgar" but I see at the bottom of the paper it was "Edsger".